code/datums/brain_damage/special.dm

```diff
@@ -0,0 +1,150 @@
+	..()
+
+/datum/brain_trauma/special/split_personality/proc/switch_personalities()
+	if(QDELETED(owner) || owner.stat == DEAD || QDELETED(stranger_backseat) || QDELETED(owner_backseat))
+		return
+
+	var/mob/living/split_personality/current_backseat
+	var/mob/living/split_personality/free_backseat
+	if(current_controller == OWNER)
+		current_backseat = stranger_backseat
+		free_backseat = owner_backseat
+	else
+		current_backseat = owner_backseat
+		free_backseat = stranger_backseat
+
+	log_game("[current_backseat]/([current_backseat.ckey]) assumed control of [owner]/([owner.ckey] due to Split Personality. (Original owner: [current_controller == OWNER ? owner.ckey : current_backseat.ckey])")
+	to_chat(owner, "<span class='userdanger'>You feel your control being taken away... your other personality is in charge now!</span>")
+	to_chat(current_backseat, "<span class='userdanger'>You manage to take control of your own body!</span>")
+
+	//Body to backseat
+
+	var/h2b_id = owner.computer_id
+	var/h2b_ip= owner.lastKnownIP
+	owner.computer_id = null
+	owner.lastKnownIP = null
+
+	free_backseat.ckey = owner.ckey
+
+	free_backseat.name = owner.name
+
+	if(owner.mind)
+		free_backseat.mind = owner.mind
+
+	if(!free_backseat.computer_id)
+		free_backseat.computer_id = h2b_id
+
+	if(!free_backseat.lastKnownIP)
+		free_backseat.lastKnownIP = h2b_ip
+
+	//Backseat to body
+
+	var/s2h_id = current_backseat.computer_id
+	var/s2h_ip= current_backseat.lastKnownIP
+	current_backseat.computer_id = null
+	current_backseat.lastKnownIP = null
+
+	owner.ckey = current_backseat.ckey
+	owner.mind = current_backseat.mind
+
+	if(!owner.computer_id)
+		owner.computer_id = s2h_id
+
+	if(!owner.lastKnownIP)
+		owner.lastKnownIP = s2h_ip
+
+	current_controller = !current_controller
+
+
+/mob/living/split_personality
+	name = "split personality"
+	real_name = "unknown conscience"
+	var/mob/living/carbon/body
+	var/datum/brain_trauma/special/split_personality/trauma
+
+/mob/living/split_personality/Initialize(mapload, _trauma)
+	if(iscarbon(loc))
+		body = loc
+		name = body.real_name
+		real_name = body.real_name
+		trauma = _trauma
+	return ..()
+
+/mob/living/split_personality/Life()
+	if(QDELETED(body))
+		qdel(src) //in case trauma deletion doesn't already do it
+
+	if((body.stat == DEAD && trauma.owner_backseat == src))
+		trauma.switch_personalities()
+		qdel(trauma)
+
+	//if one of the two ghosts, the other one stays permanently
+	if(!body.client && trauma.initialized)
+		trauma.switch_personalities()
+		qdel(trauma)
+
+	..()
+
+/mob/living/split_personality/Login()
+	..()
+	to_chat(src, "<span class='notice'>As a split personality, you cannot do anything but observe. However, you will eventually gain control of your body, switching places with the current personality.</span>")
+
+/mob/living/split_personality/say(message)
+	to_chat(src, "<span class='warning'>You cannot speak, your other self is controlling your body!</span>")
+	return FALSE
```

Comment:
Perhaps we should let the dormant personality talk to the controller?

For extra hilarity, make the controller have to talk out loud to respond.

Reply:
I pretty much took the code for this from borers, but their saycode was a copypasted horror which i decided to avoid. Also, i kinda like the idea of personalities being separated from each other completely.